otter/supervisor.py

```diff
@@ -683,3 +685,23 @@ def scrub_otter_metadata(_):
 
     d.addCallback(remove_server_from_state)
     return d
+
+
+@attributes(['scaling_group', 'server_id'])
+class EvictServerFromScalingGroup(object):
+    """
+    An Effect intent which indicates that a server should be evicted from a
+    particular group.
+    """
+
+
+@deferred_performer
+def perform_evict_server(log, transaction_id, dispatcher, intent):
```

Comment:
I assume you put `transaction_id` up here because you didn't want to pass it into the `convergence_remove_server_from_group` function? I'm wondering whether the tradeoff in complexity of setting up the dispatcher is worth it. Not really sure, I guess I'll go look in your other branch :)